core/schedule.go

```diff
@@ -27,8 +27,14 @@ import (
 	"github.com/intelsdi-x/snap/pkg/schedule"
 )
 
+// Schedule defines a scheduler.
+//
+// swagger:model Schedule
 type Schedule struct {
-	Type           string     `json:"type,omitempty"`
+	// required: true
```

Comment:
yes, schedule interval is required. Unless you find it differently, Please let me know.